src/Bicep.Core/Syntax/ImportSpecification.cs

```diff
@@ -46,35 +48,57 @@ private ImportSpecification(string name, string version, bool isValid, TextSpan
 
         public static ImportSpecification From(SyntaxBase specificationSyntax)
         {
-            switch (specificationSyntax)
+            return specificationSyntax switch
             {
-                case StringSyntax stringSyntax when stringSyntax.TryGetLiteralValue() is { } value:
-                    var (name, version, isValid) = Parse(value);
-                    var span = isValid ? new TextSpan(stringSyntax.Span.Position + 1, name.Length) : stringSyntax.Span;
-
-                    return new ImportSpecification(name, version, isValid, span);
-
-                case SkippedTriviaSyntax trivia:
-                    return new ImportSpecification(trivia.TriviaName, trivia.TriviaName, false, trivia.Span);
+                StringSyntax stringSyntax when stringSyntax.TryGetLiteralValue() is { } value
+                    => CreateFromStringSyntax(stringSyntax, value),
+                SkippedTriviaSyntax trivia
+                    => new ImportSpecification(
+                        trivia.TriviaName,
+                        trivia.TriviaName,
+                        trivia.TriviaName,
+                        false,
+                        trivia.Span),
+                _
+                    => new ImportSpecification(
+                        LanguageConstants.ErrorName,
+                        LanguageConstants.ErrorName,
+                        LanguageConstants.ErrorName,
+                        false,
+                        specificationSyntax.Span)
+            };
+        }
 
-                default:
-                    return new ImportSpecification(LanguageConstants.ErrorName, LanguageConstants.ErrorName, false, specificationSyntax.Span);
+        public SyntaxBase ToPath()
+        {
+            if (!this.IsValid)
+            {
+                return new SkippedTriviaSyntax(this.Span, Enumerable.Empty<SyntaxBase>());
             }
+            return SyntaxFactory.CreateStringLiteral(this.UnexpandedPath);
         }
 
-        private static (string Name, string Version, bool IsValid) Parse(string value)
+        private static ImportSpecification CreateFromStringSyntax(StringSyntax stringSyntax, string value)
         {
-            var match = SpecificationPattern.Match(value);
-
+            var match = BareSpecification.Match(value);
             if (!match.Success)
             {
-                return (LanguageConstants.ErrorName, LanguageConstants.ErrorName, false);
+                return new(
+                 LanguageConstants.ErrorName,
+                 LanguageConstants.ErrorName,
+                 LanguageConstants.ErrorName,
+                 false,
+                 stringSyntax.Span);
             }
 
             var name = match.Groups["name"].Value;
             var version = match.Groups["version"].Value;
+            var parts = value.Split('@');
```

Comment:
Is it possible for `value` to contain multiple `@` characters?

Reply:
In principle this could happen, however the module resolution logic would yield a diagnostic.

Reply:
@anthony-c-martin - Can you please assist with the desired behavior and proposed handling because I am not sure I understand how to bring the comment to a resolution. Also, its not clear to me if this question is a blocking issue for this PR, can you clarify?